test/unit/kernel/src/KernelWithTemplateArgumentDeduction.cpp

```diff
@@ -112,64 +122,66 @@ class KernelInvocationTemplateDeductionPointerSemantics
     }
 };
 
-TEMPLATE_LIST_TEST_CASE("kernelFuntionObjectTemplateDeductionFromPointer", "[kernel]", alpaka::test::TestAccs)
-{
-    using Acc = TestType;
-    using Dim = alpaka::Dim<Acc>;
-    using Idx = alpaka::Idx<Acc>;
-
-    alpaka::test::KernelExecutionFixture<Acc> fixture(alpaka::Vec<Dim, Idx>::ones());
-
-    using Value = std::int32_t;
-    KernelInvocationTemplateDeductionPointerSemantics<Value> kernel;
-
-    Value value{};
-    Value* pointer = &value;
-    REQUIRE(fixture(kernel, pointer, pointer));
-}
-
-TEMPLATE_LIST_TEST_CASE("kernelFuntionObjectTemplateDeductionFromPointerToConst", "[kernel]", alpaka::test::TestAccs)
-{
-    using Acc = TestType;
-    using Dim = alpaka::Dim<Acc>;
-    using Idx = alpaka::Idx<Acc>;
-
-    alpaka::test::KernelExecutionFixture<Acc> fixture(alpaka::Vec<Dim, Idx>::ones());
-
-    using Value = std::int32_t;
-    KernelInvocationTemplateDeductionPointerSemantics<Value const, Value> kernel;
-
-    Value const constValue{};
-    Value const* pointerToConst = &constValue;
-    REQUIRE(fixture(kernel, pointerToConst, pointerToConst));
-}
-
-TEMPLATE_LIST_TEST_CASE("kernelFuntionObjectTemplateDeductionFromStaticArray", "[kernel]", alpaka::test::TestAccs)
-{
-    using Acc = TestType;
-    using Dim = alpaka::Dim<Acc>;
-    using Idx = alpaka::Idx<Acc>;
-
-    alpaka::test::KernelExecutionFixture<Acc> fixture(alpaka::Vec<Dim, Idx>::ones());
-
-    using Value = std::int32_t;
-    KernelInvocationTemplateDeductionPointerSemantics<Value> kernel;
-
-    Value staticArray[4] = {};
-    REQUIRE(fixture(kernel, staticArray, staticArray));
-}
-
-TEMPLATE_LIST_TEST_CASE("kernelFuntionObjectTemplateDeductionFromConstStaticArray", "[kernel]", alpaka::test::TestAccs)
-{
-    using Acc = TestType;
-    using Dim = alpaka::Dim<Acc>;
-    using Idx = alpaka::Idx<Acc>;
-
-    alpaka::test::KernelExecutionFixture<Acc> fixture(alpaka::Vec<Dim, Idx>::ones());
-
-    using Value = std::int32_t;
-    KernelInvocationTemplateDeductionPointerSemantics<Value const, Value> kernel;
-
-    Value const constStaticArray[4] = {};
-    REQUIRE(fixture(kernel, constStaticArray, constStaticArray));
-}
+// TEMPLATE_LIST_TEST_CASE("kernelFuntionObjectTemplateDeductionFromPointer", "[kernel]", alpaka::test::TestAccs)
```

Comment:
What is happening here?

Reply:
Well, since we do no longer allow pointers in the kernel interface, I cannot compile these tests. Since these pointers do not point into buffers, an `Accessor` is also not the right tool for the job. So I commented the tests out. But we can discuss what to do about them in the future.
For example: I think we should still allow pointers at the kernel interface and remove the `static_assert` in the kernel traits.

Reply:
Yes, we should remove the `static_assert`. It is good to have it during development to see all the places where we are missing good buffer handling but we should not limit users who may want to pass raw pointers.

Reply:
Some people may not want to be compatible with SYCL.
We support a CUDA only mode for Co. patibilty with plainCUDA code for the same reason.